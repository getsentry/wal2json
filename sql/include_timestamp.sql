\set VERBOSITY terse

-- predictability
SET synchronous_commit = on;

DROP TABLE IF EXISTS tbl;
CREATE TABLE tbl (id int);

SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'wal2json');

-- One row should have three records and one timestmap
INSERT INTO tbl VALUES (1);
SELECT count(*) = 3, count(distinct ((data::json)->'timestamp')::text) = 1 FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-timestamp', '1');

-- Two rows should have six records and two timestamps
INSERT INTO tbl VALUES (2);
INSERT INTO tbl VALUES (3);
SELECT count(*) = 6, count(distinct ((data::json)->'timestamp')::text) = 2 FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-timestamp', '1');

-- Two rows in one transaction should have four records and one timestamp
INSERT INTO tbl VALUES (4), (5);
SELECT count(*) = 4, count(distinct ((data::json)->'timestamp')::text) = 1 FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-timestamp', '1');

SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');

