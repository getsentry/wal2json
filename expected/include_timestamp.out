\set VERBOSITY terse
-- predictability
SET synchronous_commit = on;
DROP TABLE IF EXISTS tbl;
NOTICE:  table "tbl" does not exist, skipping
CREATE TABLE tbl (id int);
SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'wal2json');
 ?column? 
----------
 init
(1 row)

-- One row should have three records and one timestmap
INSERT INTO tbl VALUES (1);
SELECT count(*) = 3, count(distinct ((data::json)->'timestamp')::text) = 1 FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-timestamp', '1');
 ?column? | ?column? 
----------+----------
 t        | t
(1 row)

-- Two rows should have six records and two timestamps
INSERT INTO tbl VALUES (2);
INSERT INTO tbl VALUES (3);
SELECT count(*) = 6, count(distinct ((data::json)->'timestamp')::text) = 2 FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-timestamp', '1');
 ?column? | ?column? 
----------+----------
 t        | t
(1 row)

-- Two rows in one transaction should have four records and one timestamp
INSERT INTO tbl VALUES (4), (5);
SELECT count(*) = 4, count(distinct ((data::json)->'timestamp')::text) = 1 FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-timestamp', '1');
 ?column? | ?column? 
----------+----------
 t        | t
(1 row)

SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
 ?column? 
----------
 stop
(1 row)

